Click 8.4.0 breaks type checks

**Expected behavior**
Prior to May 17, type checks on the repo succeeded

**Actual behavior (describe the problem)**
After May 17, type checks failed, due to the [new release of click 8.4.0](https://click.palletsprojects.com/en/stable/changes/#version-8-4-0), with the following errors:
```
planet/cli/types.py:27: error: Return type "list[str]" of "convert" incompatible with return type "str" in supertype "click.types.StringParamType"  [override]
planet/cli/types.py:27: error: Return type "list[str]" of "convert" incompatible with return type "str" in supertype "click.types.ParamType"  [override]
planet/cli/types.py:49: error: Return type "list[float]" of "convert" incompatible with return type "str" in supertype "click.types.StringParamType"  [override]
planet/cli/types.py:49: error: Return type "list[float]" of "convert" incompatible with return type "str" in supertype "click.types.ParamType"  [override]
```

**Related Issues**
First noticed in #1215

**Workaround**
Force click<=8.4.0

**Environment Information**
Observed in multiple environments: local (MacOS), CI.

**Installation Method**
From source

--- a/planet/cli/types.py
+++ b/planet/cli/types.py
@@ -21,14 +21,15 @@ import click
 from planet import exceptions, io
 
 
-class CommaSeparatedString(click.types.StringParamType):
+class CommaSeparatedString(click.ParamType):
     """A list of strings that is extracted from a comma-separated string."""
+    name = 'VALUE'
 
     def convert(self, value, param, ctx) -> List[str]:
         if isinstance(value, list):
             convlist = value
         else:
-            convstr = super().convert(value, param, ctx)
+            convstr = click.STRING.convert(value, param, ctx)
 
             if convstr == '':
                 self.fail('Entry cannot be an empty string.')
@@ -42,7 +43,7 @@ class CommaSeparatedString(click.types.StringParamType):
         return convlist
 
 
-class CommaSeparatedFloat(click.types.StringParamType):
+class CommaSeparatedFloat(click.ParamType):
     """A list of floats that is extracted from a comma-separated string."""
     name = 'VALUE'
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -7,7 +7,7 @@ name = "planet"
 authors = [{ name = "Planet", email = "python-sdk-contributors@planet.com" }]
 description = "Planet SDK for Python"
 dependencies = [
-  "click (>=8.0,!=8.2.1,<8.4.0)",
+  "click (>=8.0,!=8.2.1)",
   "geojson",
   "httpx>=0.28.0",
   "jsonschema",

